docs/source/examples/md/sec_agg_example.md

```diff
@@ -0,0 +1,195 @@
+
+```
+sh run.sh
+```
+
+#### How to work in CompileTime
+
+In this example, we implement compile time programs in fl_master.py
+
+```
+python fl_master.py
+```
+
+In fl_master.py, we first define FL-Strategy, User-Defined-Program and Distributed-Config. Then FL-Job-Generator generate FL-Job for federated server and worker.
+
+```python
+def linear_regression(self, inputs, label):
+        param_attrs = fluid.ParamAttr(
+            name="fc_0.b_0",
+            initializer=fluid.initializer.ConstantInitializer(0.0))
+        param_attrs = fluid.ParamAttr(
+            name="fc_0.w_0",
+            initializer=fluid.initializer.ConstantInitializer(0.0))
+        self.predict = fluid.layers.fc(input=inputs, size=10, act='softmax', param_attr=param_attrs)
+        self.sum_cost = fluid.layers.cross_entropy(input=self.predict, label=label)
+        self.loss = fluid.layers.mean(self.sum_cost)
+        self.accuracy = fluid.layers.accuracy(input=self.predict, label=label)
+        self.startup_program = fluid.default_startup_program()
+
+
+inputs = fluid.layers.data(name='x', shape=[1, 28, 28], dtype='float32')
+label = fluid.layers.data(name='y', shape=[1], dtype='int64')
+
+model = Model()
+model.linear_regression(inputs, label)
+
+job_generator = JobGenerator()
+optimizer = fluid.optimizer.SGD(learning_rate=0.01)
+job_generator.set_optimizer(optimizer)
+job_generator.set_losses([model.loss])
+job_generator.set_startup_program(model.startup_program)
+job_generator.set_infer_feed_and_target_names(
+    [inputs.name, label.name], [model.loss.name])
+
+build_strategy = FLStrategyFactory()
+build_strategy.sec_agg = True
+build_strategy.inner_step = 10
+strategy = build_strategy.create_fl_strategy()
+
+# endpoints will be collected through the cluster
+# in this example, we suppose endpoints have been collected
+endpoints = ["127.0.0.1:8181"]
+output = "fl_job_config"
+job_generator.generate_fl_job(
+    strategy, server_endpoints=endpoints, worker_num=2, output=output)
+
+```
+
+How to work in RunTime
+
+```shell
+python3 -u fl_server.py >server0.log &
+python3 -u fl_trainer.py 0 data/ >trainer0.log &
+python3 -u fl_trainer.py 1 data/ >trainer1.log &
+```
+
+In fl_server.py, we load and run the FL server job.  
+
+```
+server = FLServer()
+server_id = 0
+job_path = "fl_job_config"
+job = FLRunTimeJob()
+job.load_server_job(job_path, server_id)
+server.set_server_job(job)
+server.start()
+```
+
+In fl_trainer.py, we prepare the MNIST dataset, load and run the FL trainer job, then evaluate the accuracy.  Before training , we first prepare the party's private key and other party's public key. Then, each party generates mask using Diffie-Hellman key aggregate protocal with its parivate key and other's public key [1], and masks the model parameters before uploading to the server. Finally, the server can remove the masks by aggregating  the parameters from all the parties.
```

Comment:
protocal -> protocol
parivate -> private
-----------
Finally, the server can remove the masks by aggregating  the parameters from all the parties.
-----------
I think how the server aggregates parameters from all parties should be clarified.

Reply:
Thanks. I have corrected the typos and clarified that how the server aggregates parameters from all parties.